src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -395,7 +396,97 @@ struct KVAxesPosition {
+            std::shared_ptr<ov::Node> matched_node_last_op = nullptr;
+            if (node_to_output.count(matmul_add)) {
+                matched_node_last_op = node_to_output.at(matmul_add).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_transpose)) {
+                matched_node_last_op = node_to_output.at(matmul_transpose).get_node_shared_ptr();
+            } else if (node_to_output.count(matmul_multiply)) {
+                matched_node_last_op = node_to_output.at(matmul_multiply).get_node_shared_ptr();
+            } else {
+                matched_node_last_op = matched_node_matmul;
+            }
+            auto matched_node_result = node_to_output.at(res).get_node_shared_ptr();
+
+            auto matched_slice = std::static_pointer_cast<ov::op::v8::Slice>(matched_node_slice);
+            auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(matched_node_matmul);
+            auto matched_result = std::static_pointer_cast<ov::op::v0::Result>(matched_node_result);
+
+            // Cut original model after Slice Op:
+            matched_result->input(0).replace_source_output(matched_slice);
+
+            // Create an additional model after Slice Op:
+            auto new_param = std::make_shared<ov::op::v0::Parameter>(matched_slice->output(0).get_element_type(),
+                                                                     matched_slice->output(0).get_partial_shape());
+            matched_matmul->input(0).replace_source_output(new_param);
+            auto new_result = std::make_shared<ov::op::v0::Result>(matched_node_last_op);
+            separated_tail_protocol->first = ov::OutputVector{new_result->output(0)};
+            separated_tail_protocol->second = ov::ParameterVector{new_param};
+           
+            return true;
+        };
+        register_matcher(std::make_shared<opp::Matcher>(res, "CutTailAfterSlice"), std::move(callback));
+
+    }
+};
+
+std::shared_ptr<ov::Model> cut_tail_after_slice(std::shared_ptr<ov::Model>& model) {
+    ov::pass::GraphRewrite rewr;
+    std::shared_ptr<std::pair<ov::OutputVector, ov::ParameterVector>> tail_protocol =
+        std::make_shared<std::pair<ov::OutputVector, ov::ParameterVector>>();
+    rewr.add_matcher<CutTailAfterSlice>(tail_protocol);
+    rewr.run_on_model(model);
+
+    std::shared_ptr<ov::Model> tail_vocab_mm = nullptr;
+    if (!tail_protocol->first.empty() && !tail_protocol->second.empty()) {
+        tail_vocab_mm = std::make_shared<ov::Model>(tail_protocol->first,
+                                                    tail_protocol->second,
+                                                    "TailVocabMatmul");
+        tail_vocab_mm->validate_nodes_and_infer_types();
+        tail_vocab_mm->input(0).add_names({"output_embed"});
+        for (auto&& o : model->outputs()) {
+            if (o.get_any_name() == "logits") {
+                o.set_names({"output_embed"});
+            }
+        }
```

Comment:
this doesn't sound right, you have the exact cut point (your old `Result`) that you redirected. You can have it referenced in your struct as well (your pass knows it) and set the tensor name(s) to the right node precisely.

Also, why should this be called "output_embed"?

Reply:
Because it should be the output embedding, context-rich embedding of the last token of size [1, embed size], that got multiplied by vocabulary of size [tokens x embed size].T so we get probability logits in the end?

Reply:
is this name mandatory or some convention?

Reply:
Fixed